core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -37,127 +37,44 @@
    */
   class Factory {
-    private static final Cache UFS_CACHE = new Cache();
-
     private Factory() {} // prevent instantiation
 
     /**
-     * A class used to cache UnderFileSystems.
-     */
-    @ThreadSafe
-    private static final class Cache {
-      /**
-       * Maps from {@link Key} to {@link UnderFileSystem} instances.
-       */
-      private final ConcurrentHashMap<Key, UnderFileSystem> mUnderFileSystemMap =
-          new ConcurrentHashMap<>();
-
-      private Cache() {}
-
-      /**
-       * Gets a UFS instance from the cache if exists. Otherwise, creates a new instance and adds
-       * that to the cache.
-       *
-       * @param path the ufs path
-       * @param ufsConf the ufs configuration
-       * @return the UFS instance
-       */
-      UnderFileSystem get(String path, Object ufsConf) {
-        Key key = new Key(new AlluxioURI(path));
-        UnderFileSystem cachedFs = mUnderFileSystemMap.get(key);
-        if (cachedFs != null) {
-          return cachedFs;
-        }
-        UnderFileSystem fs = UnderFileSystemRegistry.create(path, ufsConf);
-        cachedFs = mUnderFileSystemMap.putIfAbsent(key, fs);
-        if (cachedFs == null) {
-          return fs;
-        }
-        try {
-          fs.close();
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-        return cachedFs;
-      }
-
-      void clear() {
-        mUnderFileSystemMap.clear();
-      }
-    }
-
-    /**
-     * The key of the UFS cache.
-     */
-    private static class Key {
-      private final String mScheme;
-      private final String mAuthority;
-
-      Key(AlluxioURI uri) {
-        mScheme = uri.getScheme() == null ? "" : uri.getScheme().toLowerCase();
-        mAuthority = uri.getAuthority() == null ? "" : uri.getAuthority().toLowerCase();
-      }
-
-      @Override
-      public int hashCode() {
-        return Objects.hashCode(mScheme, mAuthority);
-      }
-
-      @Override
-      public boolean equals(Object object) {
-        if (object == this) {
-          return true;
-        }
-
-        if (!(object instanceof Key)) {
-          return false;
-        }
-
-        Key that = (Key) object;
-        return Objects.equal(mScheme, that.mScheme)
-            && Objects.equal(mAuthority, that.mAuthority);
-      }
-
-      @Override
-      public String toString() {
-        return mScheme + "://" + mAuthority;
-      }
-    }
-
-    /**
-     * Clears the under file system cache.
-     */
-    public static void clearCache() {
-      UFS_CACHE.clear();
-    }
-
-    /**
-     * Gets the UnderFileSystem instance according to its schema.
+     * Gets the {@link UnderFileSystem} instance according to its UFS path. This method should only
+     * be used for journal operations and tests.
      *
      * @param path the file path storing over the ufs
      * @return instance of the under layer file system
      */
     public static UnderFileSystem get(String path) {
```

Comment:
There are still ~30 files (most in tests, some in journal v0) calling those, I kinda want to avoid further increasing the footprint of this PR.

But I think, these static getters should be consolidated with `UnderFileSystemRegistry`. Or maybe simply rename `UnderFileSystemRegistry` to `UnderFileSystemFactory`. And use `UnderFileSystemFactory#create` to replace the current reference of `UnderFileSystem.Factory.get`, what do you think?